test/testsuites/csctargetnamespacetests.go

```diff
@@ -1,65 +1,51 @@
-package e2e
+package testsuites
```

Comment:
Is the pattern you are suggesting is to create a new file for each test case?

Reply:
The pattern I'm suggesting is to create a file that has a single test suite any number of test cases. Furthermore, the test suite function should be public and all test case functions should be private.

Reply:
Then it might make sense to have separate sub-packages for opsrc and csc.

Reply:
@aravindhp I sort of disagree with this, because I think it will pigenhole us into particular use cases. The right solution is to define these not by the resource we are testing, but by the individual workflow we want to test. Perhaps there can be "generic opsrc workflows" and "generic csc workflows" test groups, but consider this example:

```
Create an opsrc + all the datastore bits
Create a csc
Delete the opsrc
```
Does that go in the opsrc bucket or the csc bucket? It definitely involves domain knowledge of the csc because creating one is required, but ultimately the context of "what happens" is up to deleting an opsrc.

Another example:

```
Create opsrc + relevant bits
Delete opsrc
```

At first glance, it seems like this should be part of the opsrc tests group. But in reality, it kind of needs to be its own test suite; the statement we are making about these test groups is:

1. Set up your stuff for your test group
2. Run your test group
3. Tear down your set up through the sdk cleanup() predicate

That delete test needs to be in its own group, because unless we order the tests in a very particular way (and rely on the sdk not breaking with an unusual cleanup) we can't ensure that the opsrc created for those tests doesn't get deleted by this very test.

Reply:
Good point @kevinrizza. I withdraw my suggestion.